data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeCoercionService.java

```diff
@@ -10,18 +10,33 @@
 import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
 
 import javax.inject.Named;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
 
 @Named
 class ParseTreeCoercionService {
+    private static final Map<Class<? extends Serializable>, Function<Object, Object>> literalTypes;
+
+    static {
```

Comment:
Could this be moved to a constructor argument and generated in a bean configuration function? Static variable can make classes difficult to unit test.

Reply:
Works for me.